### ISSUE TITLE
RdfAParser cannot parse the "prefix" attribute correctly

### ISSUE THREAD

**Issue by IS4Code** [dotnetrdf/dotnetrdf#524]
Parsing the `prefix` attribute with multiple namespaces does not work; only the first prefix is correctly mapped, for example here:
```html
<!DOCTYPE html>
<html lang="en" prefix="as: https://www.w3.org/ns/activitystreams# og: https://ogp.me/ns#">
```
As far as I can tell, the issue is [here](https://github.com/dotnetrdf/dotnetrdf/blob/4c5d0495af7b9580e82eedf472fc4a8d78e8bff6/Libraries/dotNetRdf/Parsing/RdfAParserBase.cs#L1382). It moves through the string using Peek, but once a whitespace character is encountered, it doesn't skip it and then considers it a part of the prefix on the next cycle.

Preferably, the `ParsePrefixAttribute` method should ignore any additional whitespace found at the beginning or at the end of the string, and skip any consecutive whitespace between prefix definitions. Maybe this whole thing could be simplified with regex; something like `(?<prefix>[^\s]*:)\s+(?<uri>[^\s]+)` should work nicely.

Additionally, it uses `hiddenPrefixes.Add` to register the namespace (in some situations), but if the same prefix is used multiple times, this will throw an exception. I think it would be better to use the indexer and overwrite the namespace if already registered. Not sure whether this is allowed in RDFa though, but it seems like the most fool-proof option.

Also looking at `ParsePrefixAttribute`, I am not sure it works correctly with regard to `hiddenPrefixes`. It creates the dictionary if it's `null`, but it cannot set it outside the method in that case, since the dictionary reference is passed by value. If that is not intended, `ref Dictionary<string, Uri> hiddenPrefixes` should be used as the parameter, to actually initialize it if it's `null`.

Lastly, I don't agree with `if (!(uri.EndsWith("/") || uri.EndsWith("#"))) uri += "#";`. According to [the specification](https://www.w3.org/TR/rdfa-core/#h-s_curieprocessing), "The author is free to split the IRI at any point.". Not sure if omitting a `/` or `#` is a common source of errors, but in RDFa (or in any other RDF serialization language for that matter) the prefix and the local name should be concatenated as they are, without inserting any special characters.

**Comment by wiresio** [comment]
Would be great if the fix can make it to the next (bug fix) release 3.0.x. Was working in 2.7.5.

**Comment by kal** [comment]
The PR #563 addresses the issues on this ticket.
